libgit/autogit_manager.go

```diff
@@ -508,6 +528,13 @@ func (am *AutogitManager) doDelete(req deleteReq) (err error) {
 		if err == nil {
 			err = workDoneErr
 		}
+		// Remove the lock file.  This happens outside of the main
+		// deletion single-op, and so won't appear strictly atomically
+		// with the rest of the delete.
```

Comment:
Hmm I'm bit confused. It still creates the lock file through `canWorkOnRepo`, and does an unlock through `workDoneOnRepo`. Could you explain what you meant by not "strictly atomically"?

Reply:
The call to `workDoneOnRepo` unlocks the file, which causes a `FinishSingleOp` call.  So the `Remove` call below happens _after_ that first single operation finishes, and is part of a new operation.  That means the lock will be removed in a separate MD update from the rest of the deletion.

Reply:
Oh I see; makes sense. Thanks!